QUARKUS-2829.md

```diff
@@ -0,0 +1,55 @@
+# QUARKUS-2829 - Internal support for Micrometer metrics when using managed Vert.x in Quarkus
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-2829
+
+
+As part of supporting Vert.x use-cases on Quarkus we need to support collecting metrics using micrometer for the managed Vert.x instance in Quarkus. This is also a requirement for some layered products like AMQ Streams and OpenShift Serverless that today are using Vert.x, but needs to move to use Quarkus. For 2.13 the support for this is limited to OpenShift Serverless and AMQ Streams, this is not supported for general use. Support for general use should come in RHBQ 3.x
+
+## Scope of testing
+
+Managed Vert.x in Quarkus contains following extensions:
+- Messaging clients:
+AMQP,Kafka, MQTT. Extension names: quarkus-smallrye-reactive-messaging-amqp, *-kafka, *-mqtt.
+- DB clients: Mssql, DB2, Oracle, MySQL, Postgres(pg). Extension names: quarkus-reactive-(name)-client.
+- Redis client: quarkus-redis-client
+- quarkus-mailer
+- quarkus-vertx (the core API)
+
+
+Current implementation provides following metrics (full names is `poolType+".pool."+MetricName`):
+- queue.delay
+- queue.size
+- usage
+- active
+- idle
+- ratio
+- completed
+- processing
+- current
+- reset
+- pending
+- connections
+
+Ideally (for Quarkus 3.0), we need to check, that we can get metrics for all supported extensions, and for working with Vert.X specific functionality of verticles and event bus.
```

Comment:
Maybe this should be in future considerations chapter - this won't be in scope of testing for QUARKUS-2829, since that's scoped just for what you mention in the next sentence.

Reply:
Moved to "future considerations"